### PR TITLE
feat: pruning of the content-addressable store

### PR DESCRIPTION
**Pull request by zkochan**
ref #2470

### CODE REVIEW

File: packages/package-store/src/storeController/prune.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  for (const dir of dirs) {
+    const subdir = path.join(cafsDir, dir)
+    for (const fileName of await fs.readdir(subdir)) {
+      const filePath = path.join(subdir, fileName)
+      if (fileName.endsWith('-index.json')) {
+        pkgIndexFiles.push(filePath)
+        continue
+      }
+      const stat = await fs.stat(filePath)
+      if (stat.nlink === 1) {
```
Comment (KSXGitHub) [inline packages/package-store/src/storeController/prune.ts:26]: nlink can be BigInt (1n), which is not equal to 1.

File: packages/package-store/src/storeController/prune.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        await fs.unlink(filePath)
+        fileCounter++
+        removedHashes.add(ssri.fromHex(`${dir}${fileName}`, 'sha512').toString())
+      }
+    }
+  }
+  globalInfo(`Removed ${fileCounter} files`)
```
Comment (aparajita) [inline packages/package-store/src/storeController/prune.ts]:
```suggestion
  globalInfo(`Removed ${fileCounter} file${fileCounter === 1 ? '' : 's'`)
```

File: packages/package-store/src/storeController/prune.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  let pkgCounter = 0
+  for (const pkgIndexFilePath of pkgIndexFiles) {
+    const pkgFilesIndex = await loadJsonFile<object>(pkgIndexFilePath)
+    if (removedHashes.has(pkgFilesIndex['package.json'].integrity)) {
+      await fs.unlink(pkgIndexFilePath)
+      pkgCounter++
+    }
+  }
+  globalInfo(`Removed ${pkgCounter} packages`)
```
Comment (aparajita) [inline packages/package-store/src/storeController/prune.ts]:
```suggestion
  globalInfo(`Removed ${pkgCounter} package${pkgCounter === 1 ? '' : 's'}`)
```

File: packages/plugin-commands-store/test/storePrune.ts [hunk 4]
```diff
@@ -37,7 +37,7 @@ test('remove unreferenced packages', async (t) => {
 
   t.ok(reporter.calledWithMatch({
     level: 'info',
-    message: `- localhost+${REGISTRY_MOCK_PORT}/is-negative/2.1.0`,
+    message: 'Removed 1 packages',
```
Comment (aparajita) [inline packages/plugin-commands-store/test/storePrune.ts]:
```suggestion
    message: 'Removed 1 package',
```

File: packages/plugin-commands-store/test/storePrune.ts [hunk 5]
```diff
@@ -56,7 +56,7 @@ test('remove unreferenced packages', async (t) => {
 
   t.notOk(reporter.calledWithMatch({
     level: 'info',
-    message: `- localhost+${REGISTRY_MOCK_PORT}/is-negative/2.1.0`,
+    message: 'Removed 1 packages',
```
Comment (aparajita) [inline packages/plugin-commands-store/test/storePrune.ts]:
```suggestion
    message: 'Removed 1 package',
```

File: packages/package-store/src/storeController/prune.ts (hunk 1 above)
Comment (zkochan) [reply packages/package-store/src/storeController/prune.ts:26]: ok, added

**Pull request merged**